### PR TITLE
DEVEXP-544: Add Affinity to the image registry CRD

### PR DESCRIPTION
**Pull request by coreydaley**

### CODE REVIEW

File: imageregistry/v1/types.go [hunk 1]
```diff
@@ -75,18 +75,21 @@ type ImageRegistrySpec struct {
...
 	// +optional
 	Resources *corev1.ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,12,opt,name=resources"`
+	// affinity is a group of node affinity scheduling rules for the image registry pod(s).
+	// +optional
+	Affinity *corev1.Affinity `json:"affinity,omitempty" protobuf:"bytes,13,opt,name=affinity"`
 	// nodeSelector defines the node selection constraints for the registry
 	// pod.
 	// +optional
-	NodeSelector map[string]string `json:"nodeSelector,omitempty" protobuf:"bytes,13,rep,name=nodeSelector"`
+	NodeSelector map[string]string `json:"nodeSelector,omitempty" protobuf:"bytes,14,rep,name=nodeSelector"`
```
Comment (bparees) [inline imageregistry/v1/types.go]: i'm pretty sure changing these is a no-no, you need to add it to the end.

Comment (bparees) [reply imageregistry/v1/types.go]: https://developers.google.com/protocol-buffers/docs/proto3#updating

Comment (coreydaley) [reply imageregistry/v1/types.go]: Done.

**Pull request merged**